packages/ledgerstate/conflict.go

```diff
@@ -0,0 +1,459 @@
+
+// Conflict represents a set of Branches that are conflicting with each other.
+type Conflict struct {
+	id               ConflictID
+	memberCount      int
+	memberCountMutex sync.RWMutex
+
+	objectstorage.StorableObjectFlags
+}
+
+// NewConflict is the constructor for new Conflicts.
+func NewConflict(conflictID ConflictID) *Conflict {
+	return &Conflict{
+		id: conflictID,
+	}
+}
+
+// ConflictFromBytes unmarshals a Conflict from a sequence of bytes.
+func ConflictFromBytes(bytes []byte) (conflict *Conflict, consumedBytes int, err error) {
+	marshalUtil := marshalutil.New(bytes)
+	if conflict, err = ConflictFromMarshalUtil(marshalUtil); err != nil {
+		err = xerrors.Errorf("failed to parse Conflict from MarshalUtil: %w", err)
+		return
+	}
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// ConflictFromMarshalUtil unmarshals a Conflict using a MarshalUtil (for easier unmarshaling).
+func ConflictFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (conflict *Conflict, err error) {
+	conflict = &Conflict{}
+	if conflict.id, err = ConflictIDFromMarshalUtil(marshalUtil); err != nil {
+		err = xerrors.Errorf("failed to parse ConflictID from MarshalUtil: %w", err)
+		return
+	}
+	memberCount, err := marshalUtil.ReadUint64()
+	if err != nil {
+		err = xerrors.Errorf("failed to parse member count (%v): %w", err, ErrParseBytesFailed)
+		return
+	}
+	conflict.memberCount = int(memberCount)
+
+	return
+}
+
+// ConflictFromObjectStorage restores a Conflict object that was stored in the ObjectStorage.
+func ConflictFromObjectStorage(key []byte, data []byte) (outputMetadata *Conflict, err error) {
+	if outputMetadata, _, err = ConflictFromBytes(byteutils.ConcatBytes(key, data)); err != nil {
+		err = xerrors.Errorf("failed to parse Conflict from bytes: %w", err)
+		return
+	}
+
+	return
+}
+
+// ID returns the identifier of this Conflict.
+func (c *Conflict) ID() ConflictID {
+	return c.id
+}
+
+// MemberCount returns the amount of Branches that are part of this Conflict.
+func (c *Conflict) MemberCount() int {
+	c.memberCountMutex.RLock()
+	defer c.memberCountMutex.RLock()
+
+	return c.memberCount
+}
+
+// IncreaseMemberCount increase the MemberCount of this Conflict.
+func (c *Conflict) IncreaseMemberCount(optionalDelta ...int) int {
+	delta := 1
+	if len(optionalDelta) >= 1 {
+		delta = optionalDelta[0]
+	}
+
+	c.memberCountMutex.Lock()
+	defer c.memberCountMutex.Unlock()
+
+	c.memberCount = c.memberCount + delta
+	c.SetModified()
+
+	return c.memberCount
+}
+
+// DecreaseMemberCount decreases the MemberCount of this Conflict.
+func (c *Conflict) DecreaseMemberCount(optionalDelta ...int) (newMemberCount int) {
+	delta := 1
+	if len(optionalDelta) >= 1 {
+		delta = optionalDelta[0]
+	}
+
+	c.memberCountMutex.Lock()
+	defer c.memberCountMutex.Unlock()
+
+	c.memberCount = c.memberCount - delta
+	c.SetModified()
+	newMemberCount = c.memberCount
```

Comment:
Here you should be consistent in the way the method is structured. In `IncreaseMemberCount` you simply return `c.memberCount` vs here you introduce `newMemberCount`.

Reply:
okay I will add a named parameter as well as its a more dscriptive method signature then